cdflow_commands/plugins/ecs.py

```diff
@@ -410,27 +410,28 @@ class ECSMonitor:
     _TIMEOUT = 600
     _INTERVAL = 15
 
-    def __init__(self, ecs_event_iterator):
+    def __init__(self, ecs_event_iterator, plan_only=False):
         self._ecs_event_iterator = ecs_event_iterator
         self._previous_running_count = 0
+        self._plan_only = plan_only
 
     def wait(self):
-        start = time()
+        if not self._plan_only:
+            self._check_ecs_deploy_progress()
 
+    def _check_ecs_deploy_progress(self):
+        start = time()
         for event in self._ecs_event_iterator:
-            if time() - start > self._TIMEOUT:
-                raise TimeoutError(
-                    'Deployment timed out - didn\'t complete '
-                    'within {} seconds'.format(self._TIMEOUT)
-                )
-
             self._show_deployment_progress(event)
             self._check_for_failed_tasks(event)
-
             if event.done:
                 logger.info('Deployment complete')
                 return True
-
+            if time() - start > self._TIMEOUT:
```

Comment:
Good spot Bart! I thought about this for a couple of seconds and was 50/50 on whether to do it or not. First of all, the linter through a complexity error so had to abstract that part and then decided that I thought it would be good to check one last time for a failing task before throwing the timeout error (very unlikely scenario obviously).